src/osyris/spatial/subdomain.py

```diff
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/osyris-project/osyris)
+
+import numpy as np
+from .. import Dataset
+
+
+def extract_sphere(dataset, radius, origin):
+    """
+    Extract a spherical subdomain around an origin point.
+    """
+    subdomain = Dataset()
+    subdomain.meta = dataset.meta.copy()
+
+    for name, group in dataset.items():
+        pos = group.get("position", group.parent["amr"]["position"])
+        if pos.shape[0] != group.shape:
```

Comment:
This looks wrong, I think I forgot to change something about the Datagroup `shape` when i introduced the Vector class.
I will make a change so that we can just compare the shapes directly, without needing the `[0]`.

Reply:
See https://github.com/osyris-project/osyris/pull/107

Reply:
I'll push the change once #107 is merged.